daemon/logger/gelf/gelf.go

```diff
@@ -171,6 +171,10 @@ func (s *gelfLogger) Log(msg *logger.Message) error {
 		level = gelf.LOG_ERR
 	}
 
+	if len(msg.Line) == 0 {
+		msg.Line = []byte("<empty-line>")
```

Comment:
Not a fan of the `<empty-line>` magic value; I expect empty lines being important to be a bit of a corner-case, and  recall we had something similar with the Splunk logging driver; for that driver we're skipping empty lines (see https://github.com/moby/moby/pull/34520)

So, unless there's a strong case for preserving empty lines, I'd suggest skipping them entirely.

In that case, I would prefer this check to be the very first in the function (so before the `level := ....` line), and return early;

```go
if len(msg.Line) == 0 {
	return nil
}
```

Reply:
Given #34520 and no strong case for empty lines, I agree that it is better to skip them entirely. Consistency is increased between drivers and to be honest most processing logic (e.g. via Logstash) ignores them anyway.

I'll change it to your suggested version.